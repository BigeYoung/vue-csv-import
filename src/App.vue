<template>
    <div id="app">
        <img alt="Vue logo" src="./assets/logo.png">
        <div class="container">
            <div class="row">
                <div class="col-6 offset-3">
                    <form class="form">
                        <vue-csv-import v-model="csv" ref="importer" :map-fields="['date', 'name']"></vue-csv-import>
                    </form>
                    {{ csv }}
                </div>
            </div>
        </div>
    </div>
</template>

<script>
  import VueCsvImport from './components/VueCsvImport.vue';

  export default {
    name: 'app',
    components: {
      VueCsvImport
    },
    data() {
      return {
        csv: null,
      };
    }
  };
</script>

<style>
    #app {
        font-family: 'Avenir', Helvetica, Arial, sans-serif;
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
        text-align: center;
        color: #2c3e50;
        margin-top: 60px;
    }

    #app .container {
        text-align: left;
    }
</style>
