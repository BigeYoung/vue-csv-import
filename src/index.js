export { default as VueCsvImport } from './components/VueCsvImport';
