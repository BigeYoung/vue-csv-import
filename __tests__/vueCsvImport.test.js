import { VueCsvImport } from '../src';
import Vue from 'vue/dist/vue.js';

describe('VueCsvImport', () => {
    Vue.component('vue-csv-import', VueCsvImport);

    beforeEach(() => {
        document.body.innerHTML = `
            <div id="app">
                <vue-csv-import :map-fields="['one', 'two', 'three']" url="/hello"></vue-csv-import>
            </div>
        `;
    });

    it('can mount', async () => {
        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });
});

async function createVm() {
    const vm = new Vue({
        el: '#app',
    });

    await Vue.nextTick(() => {});

    return { app: vm, component: vm.$children[0] };
}
