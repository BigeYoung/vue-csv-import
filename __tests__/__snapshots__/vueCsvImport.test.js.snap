// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueCsvImport can mount 1`] = `
"
            <div id=\\"app\\"><div class=\\"vue-csv-uploader\\"><div class=\\"form\\"><div class=\\"form-check\\"><input type=\\"checkbox\\" id=\\"hasHeaders\\" class=\\"form-check-input\\"> <label for=\\"hasHeaders\\" class=\\"form-check-label\\">
                File Has Headers
            </label></div> <div class=\\"form-group\\"><input type=\\"file\\" name=\\"csv\\" class=\\"form-control\\"></div> <div class=\\"form-group\\"><input type=\\"submit\\" class=\\"btn btn-default\\"></div> <!----></div></div></div>
        "
`;
